common/sgx/edl/attestation.edl

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/*
+**==============================================================================
+**
+** sgx/attestation.edl:
+**
+**     Internal ECALLs/OCALLs to be used by liboehost/liboecore for SGX-specific
+**     attestation.
+**
+**==============================================================================
+*/
+
+enclave
+{
+    include "openenclave/bits/types.h"
+
+    // Most types in sgxtypes.h are packed to ensure the memory layout is
+    // the same as data retrieved from the CPU. Since EDL does not support
+    // packed structures, these types cannot be defined in EDL. They do not
+    // need the benefit of deep copy because they are all laid out in
+    // contiguous memory.
+    include "openenclave/bits/sgx/sgxtypes.h"
+
+    trusted
+    {
+        public oe_result_t oe_get_sgx_report_ecall(
+            [in, size=opt_params_size] const void* opt_params,
+            size_t opt_params_size,
+            [out] sgx_report_t* report);
+    };
+
+    untrusted
+    {
+        oe_result_t oe_get_qetarget_info_ocall(
```

Comment:
I agree that we should do this but I would rather do it in a separate PR (disambiguate all the SGX EDL function names) because it will touch a lot of code, some of which is in flux.

This PR doesn't actually publish the EDL files yet (just prepares them to be published) so we still have time to change before they're published.